task.ps1

```diff
@@ -21,3 +22,28 @@ $nsgRuleHTTP = New-AzNetworkSecurityRuleConfig -Name HTTP  -Protocol Tcp -Direct
 New-AzNetworkSecurityGroup -Name $networkSecurityGroupName -ResourceGroupName $resourceGroupName -Location $location -SecurityRules $nsgRuleSSH, $nsgRuleHTTP
 
 # ↓↓↓ Write your code here ↓↓↓
+Write-Host "Creating a virtual network $virtualNetworkName and a subnet $subnetName ..."
+New-AzVirtualNetwork -Name $virtualNetworkName -ResourceGroupName $resourceGroupName -Location $location -AddressPrefix $vnetAddressPrefix -Subnet @(New-AzVirtualNetworkSubnetConfig -Name $subnetName -AddressPrefix $subnetAddressPrefix)
+
+Write-Host "Creating a public IP address $publicIpAddressName with DNS label ..."
+New-AzPublicIpAddress -ResourceGroupName $resourceGroupName -Location $location -Name $publicIpAddressName -AllocationMethod Static -Sku Basic -DomainNameLabel $domainNameLabel
```

Comment:
The `New-AzPublicIpAddress` command uses a static allocation method. Ensure that this is intentional, as it might incur additional costs compared to dynamic allocation.